### PR TITLE
fix block mask typing

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Super nit

### CODE REVIEW

**Review by Cyrilvallez — APPROVED** [review]

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -849,13 +849,13 @@ def _load_state_dict_into_meta_model(
...
-        if serialized_param_name not in expected_keys:
-            continue
-
         # serialized_param_name is the raw, serialized name
         # fixed_param_name is the model's equivalent
         fixed_param_name, _ = model.rename_key(serialized_param_name)
 
+        if fixed_param_name not in expected_keys:
+            continue
+
```
Comment (zucchini-nlp) [inline src/transformers/modeling_utils.py:858]: BTW, this was done on purpose, to load the `Gemma3CausalLM` with accelerate. Before the TP loading was updated, we cropped prefix only after checking existence in expected keys

Comment (Cyrilvallez) [reply src/transformers/modeling_utils.py:858]: For model actually modifying the keys, checking the serialized keys will basically skip everything 😅 Anyway, we will merge https://github.com/huggingface/transformers/pull/36033 which will use a single key renaming at the very beginning to avoid these issues

**Review by LysandreJik — APPROVED** [review]
Ok good!
